src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Folding/FoldingRangeEndpoint.cs

```diff
@@ -24,28 +20,22 @@ namespace Microsoft.AspNetCore.Razor.LanguageServer.Folding
 {
     internal class FoldingRangeEndpoint : IVSFoldingRangeEndpoint
     {
+        private readonly DocumentContextFactory _documentContextFactory;
         private readonly RazorDocumentMappingService _documentMappingService;
-        private readonly ProjectSnapshotManagerDispatcher _projectSnapshotManagerDispatcher;
-        private readonly DocumentResolver _documentResolver;
         private readonly ClientNotifierServiceBase _languageServer;
-        private readonly DocumentVersionCache _documentVersionCache;
         private readonly IEnumerable<RazorFoldingRangeProvider> _foldingRangeProviders;
         private readonly ILogger _logger;
 
         public FoldingRangeEndpoint(
+            DocumentContextFactory documentContextFactory,
             RazorDocumentMappingService documentMappingService,
-            ProjectSnapshotManagerDispatcher projectSnapshotManagerDispatcher,
-            DocumentResolver documentResolver,
             ClientNotifierServiceBase languageServer,
-            DocumentVersionCache documentVersionCache,
             IEnumerable<RazorFoldingRangeProvider> foldingRangeProviders,
             ILoggerFactory loggerFactory)
         {
+            _documentContextFactory = documentContextFactory ?? throw new ArgumentNullException(nameof(documentContextFactory));
             _documentMappingService = documentMappingService ?? throw new ArgumentNullException(nameof(documentMappingService));
             _languageServer = languageServer ?? throw new ArgumentNullException(nameof(languageServer));
-            _projectSnapshotManagerDispatcher = projectSnapshotManagerDispatcher ?? throw new ArgumentNullException(nameof(projectSnapshotManagerDispatcher));
```

Comment:
This file is a great example of all the content we can get rid of in an effort to utilize the document context concept